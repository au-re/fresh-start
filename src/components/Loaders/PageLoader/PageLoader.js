import { Spin, message } from "antd";
import React, { Component } from "react";

/**
 * Component that renders while a page loads
 *
 * @class PageLoader
 * @extends {Component}
 */
class PageLoader extends Component {
  render() {
    const { className, error, pastDelay } = this.props;
    if (error) {
      message.error("something went wrong");
      return null;
    }
    if (pastDelay) {
      return <div className={className}>
        <Spin size="large" />
      </div>;
    }
    return null;
  }
}

export default PageLoader;
