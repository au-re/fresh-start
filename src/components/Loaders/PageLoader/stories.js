import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import React from "react";

import PageLoader from "./index";

storiesOf("PageLoader", module)

  .add("basic usage", withInfo(`Placeholder while a page is loading`)(
    () => <PageLoader pastDelay />))

  .add("on error", withInfo(`Placeholder displays an error on failure`)(
    () => <PageLoader error />));
