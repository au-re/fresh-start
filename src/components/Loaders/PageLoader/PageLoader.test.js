import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";

import PageLoader from "./PageLoader";

Enzyme.configure({ adapter: new Adapter() });

describe("PageLoader", () => {

  it("renders without props", () => {
    expect(<PageLoader />).toMatchSnapshot();
  });
});
