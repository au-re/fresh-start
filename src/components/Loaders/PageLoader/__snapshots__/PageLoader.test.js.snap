// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoader renders without props 1`] = `<PageLoader />`;
